src/uct/ib/ud/ud_inl.h

```diff
@@ -66,25 +66,38 @@ static inline uct_ud_send_skb_t *uct_ud_iface_get_tx_skb(uct_ud_iface_t *iface,
         return NULL;
     }
 
-    if (ep->tx.psn == ep->tx.max_psn) {
+    if (ucs_unlikely(ep->tx.psn == ep->tx.max_psn)) {
```

Comment:
you are using plain == to compare psn's here, and PSN_COMPARE macro in a previous case